lib/Extension/Navigation/Tests/IntegrationTestCase.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Phpactor\Extension\Navigation\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Phpactor\Container\Container;
+use Phpactor\Container\PhpactorContainer;
+use Phpactor\Extension\ClassToFile\ClassToFileExtension;
+use Phpactor\Extension\CodeTransformExtra\CodeTransformExtraExtension;
+use Phpactor\Extension\CodeTransform\CodeTransformExtension;
+use Phpactor\Extension\ComposerAutoloader\ComposerAutoloaderExtension;
+use Phpactor\Extension\Core\CoreExtension;
+use Phpactor\Extension\FilePathResolver\FilePathResolverExtension;
+use Phpactor\Extension\Logger\LoggingExtension;
+use Phpactor\Extension\Navigation\NavigationExtension;
+use Phpactor\Extension\Php\PhpExtension;
+use Phpactor\Extension\SourceCodeFilesystem\SourceCodeFilesystemExtension;
+use Phpactor\Extension\WorseReflection\WorseReflectionExtension;
+use Phpactor\TestUtils\Workspace;
+
+abstract class IntegrationTestCase extends TestCase
+{
+    protected Workspace $workspace;
+
+    public function setUp(): void
+    {
+        $this->workspace = $this->workspace();
+        $this->workspace->reset();
+    }
+
+    /**
+     * @param array{
+     * 'navigator.destinations': array<string, string>,
+     * 'navigator.autocreate': array<string, string>,
+     * } $config
+     */
+    protected function container(array $config): Container
+    {
+        $key = serialize($config);
+        static $container = [];
+
+        if (isset($container[$key])) {
+            return $container[$key];
+        }
+
+        $container[$key] = PhpactorContainer::fromExtensions([
+            CodeTransformExtension::class,
+            CodeTransformExtraExtension::class,
+            PhpExtension::class,
+            CoreExtension::class,
+            NavigationExtension::class,
+            LoggingExtension::class,
+            SourceCodeFilesystemExtension::class,
+            ClassToFileExtension::class,
+            ComposerAutoloaderExtension::class,
+            FilePathResolverExtension::class,
+            WorseReflectionExtension::class,
+        ], array_merge([
+            LoggingExtension::PARAM_ENABLED=> true,
```

Comment:
Either disable or examine why this warning is even produced: https://github.com/phpactor/phpactor/actions/runs/11846687499/job/33014838113?pr=2776#step:5:235

Reply:
because the COMPOSER_AUTOLOADER path is based on the APPLICATION_ROOT perhaps?

The reason these integration test cases all over the place is because these were originally separate packages. At some level it would make sense to have a single "Phpactor" IntegrationTestCase providing the workspace + container, or maybe a better way to bootstrap it.

Reply:
In `IntegrationTestCase` I disabled composer extension